packages/plugin-hardhat/src/force-import.ts

```diff
@@ -30,28 +29,29 @@ export interface ForceImportFunction {
 
 export function makeForceImport(hre: HardhatRuntimeEnvironment): ForceImportFunction {
   return async function forceImport(
-    proxyOrBeacon: ContractAddressOrInstance,
+    addressOrInstance: ContractAddressOrInstance,
     ImplFactory: ContractFactory,
     opts: Options = {},
   ) {
     const { provider } = hre.network;
     const manifest = await Manifest.forNetwork(provider);
 
-    const proxyOrBeaconAddress = getContractAddress(proxyOrBeacon);
+    const address = getContractAddress(addressOrInstance);
 
-    const implAddress = await getImplementationAddressFromProxy(provider, proxyOrBeaconAddress);
+    const implAddress = await getImplementationAddressFromProxy(provider, address);
     if (implAddress !== undefined) {
-      await importProxyToManifest(provider, hre, proxyOrBeaconAddress, implAddress, ImplFactory, opts, manifest);
+      await importProxyToManifest(provider, hre, address, implAddress, ImplFactory, opts, manifest);
 
-      return ImplFactory.attach(proxyOrBeaconAddress);
-    } else if (await isBeacon(provider, proxyOrBeaconAddress)) {
-      const beaconImplAddress = await getImplementationAddressFromBeacon(provider, proxyOrBeaconAddress);
+      return ImplFactory.attach(address);
+    } else if (await isBeacon(provider, address)) {
+      const beaconImplAddress = await getImplementationAddressFromBeacon(provider, address);
       await addImplToManifest(hre, beaconImplAddress, ImplFactory, opts);
 
       const UpgradeableBeaconFactory = await getUpgradeableBeaconFactory(hre, ImplFactory.signer);
-      return UpgradeableBeaconFactory.attach(proxyOrBeaconAddress);
+      return UpgradeableBeaconFactory.attach(address);
     } else {
```

Comment:
Do we at least know whether there is a contract at this address? Not sure if that's something we should require but it seems like a reasonable sanity check.